db/table_cache.cc

```diff
@@ -426,13 +433,15 @@ Status TableCache::Get(
   IterKey row_cache_key;
   std::string row_cache_entry_buffer;
 
-  // Check row cache if enabled. Since row cache does not currently store
-  // sequence numbers, we cannot use it if we need to fetch the sequence.
+  // Check row cache if enabled.
+  // Reuse row_cache_key sequence number when row cache hits.
   if (ioptions_.row_cache && !get_context->NeedToReadSequence()) {
     auto user_key = ExtractUserKey(k);
-    CreateRowCacheKeyPrefix(options, fd, k, get_context, row_cache_key);
-    done = GetFromRowCache(user_key, row_cache_key, row_cache_key.Size(),
-                           get_context);
+    uint64_t cache_entry_seq_no =
+        CreateRowCacheKeyPrefix(options, fd, k, get_context, row_cache_key);
+    done = GetFromRowCache(
+        user_key, row_cache_key, row_cache_key.Size(), get_context,
+        cache_entry_seq_no == 0 ? 0 : cache_entry_seq_no - 1);
     if (!done) {
```

Comment:
Do you think it would make sense to move this logic to be within `GetFromRowCache`? I mean to pass `seq_no` as is here and when passing the sequence number to `replayGetContextLog`, we add this conditional logic. Because at the row cache level, `Insert` and `Lookup` are still saving and retrieving based on the same sequence number, adding the conditional logic here makes it look like this `-1` operation is involved in cache `Lookup` operation, which is not the case. 

It would be great to add some documentation to explain the line that does this `-1` operation too. Something like : successful retrieving row from row cache guarantees visibility of the entries w.r.t snapshot, the saved sequence number is `-1` before passing to `replayGetContextLog` to ensure the ReadCallback check passes. This is needed because of the mirroring `+1` operation in `CreateRowCacheKeyPrefix`.